### PR TITLE
Package Type API

### PR DESCRIPTION
**Pull request by martinmaly**
Add PackageRevision `type` field to capture whether the packge is draft,
proposed, of final.

### CODE REVIEW

File: porch/api/porch/types_packagerevisions.go [hunk 1]
```diff
@@ -40,14 +40,22 @@ type PackageRevisionList struct {
...
+	PackageRevisionTypeFinal    PackageRevisionType = "final"
+)
 
+// PackageRevisionSpec defines the desired state of PackageRevision
+type PackageRevisionSpec struct {
+	PackageName    string `json:"packageName,omitempty"`
+	Revision       string `json:"revision,omitempty"`
 	RepositoryName string `json:"repository,omitempty"`
 
+	Type PackageRevisionType `json:"type,omitempty"`
```
Comment (justinsb) [inline porch/api/porch/types_packagerevisions.go:57]: Let's not change it in this PR (I assume it'll break the whole chain), but this reads more like `state` or `lifecycle` than `type` to me.

File: porch/api/porch/v1alpha1/types_packagerevisions.go [hunk 2]
```diff
@@ -41,14 +41,22 @@ type PackageRevisionList struct {
...
 }
 
-// PackageRevisionSpec defines the desired state of PackageRevision
-type PackageRevisionSpec struct {
-	PackageName string `json:"packageName,omitempty"`
+type PackageRevisionType string
 
-	Revision string `json:"revision,omitempty"`
+const (
+	PackageRevisionTypeDraft    PackageRevisionType = "draft"
```
Comment (justinsb) [inline porch/api/porch/v1alpha1/types_packagerevisions.go:47]: I _think_ we generally capitalize user-visible "enums" (e.g. Draft, Proposed, Final)

**Review by justinsb — APPROVED** [review]

File: porch/api/porch/types_packagerevisions.go (hunk 1 above)
Comment (martinmaly) [reply porch/api/porch/types_packagerevisions.go]: I decided to do this in this PR ... there will be potential for less long-term fallout. Thanks!

File: porch/api/porch/v1alpha1/types_packagerevisions.go (hunk 2 above)
Comment (martinmaly) [reply porch/api/porch/v1alpha1/types_packagerevisions.go]: Looking at prior art, this is correct. I fixed these constants but there are more to fix too that we currently lower-case. I'll add tracking issue for that.

**Pull request merged**